src/packages/@ncigdc/modern_components/ClinicalAnalysis/ClinicalVariableCard/CategoricalVariableCard.js

```diff
@@ -0,0 +1,1060 @@
+
+                <Dropdown
+                  button={(
+                    <Button
+                      rightIcon={<DownCaretIcon />}
+                      style={{
+                        ...visualizingButton,
+                        padding: '0 12px',
+                      }}
+                      >
+                      Customize Bins
+                    </Button>
+                  )}
+                  dropdownStyle={{ right: 0 }}
+                  >
+                  <DropdownItem
+                    onClick={() => dispatch(setModal(
+                      <GroupValuesModal
+                        bins={variable.bins}
+                        dataBuckets={dataBuckets}
+                        fieldName={humanify({ term: fieldName })}
+                        modalStyle={{
+                          maxWidth: '720px',
+                          width: '90%',
+                        }}
+                        onClose={() => dispatch(setModal(null))}
+                        onUpdate={(newBins) => {
+                          dispatch(
+                            updateClinicalAnalysisVariable({
+                              fieldName,
+                              id,
+                              value: newBins,
+                              variableKey: 'bins',
+                            }),
+                          );
+                          dispatch(setModal(null));
+                        }}
+                        />
+                    ))}
+                    style={styles.actionMenuItem}
+                    >
+                    Edit Bins
+                  </DropdownItem>
+
+                  <DropdownItem
+                    onClick={() => {
+                      if (resetBinsDisabled) return;
+                      dispatch(
+                        updateClinicalAnalysisVariable({
+                          fieldName,
+                          id,
+                          value: dataBuckets.reduce((acc, r) => ({
+                            ...acc,
+                            [r.key]: {
+                              ...r,
+                              groupName: r.key,
+                            },
+                          }), {}),
+                          variableKey: 'bins',
+                        }),
+                      );
+                    }}
+                    style={{
+                      ...styles.actionMenuItem,
+                      ...(resetBinsDisabled
+                        ? styles.actionMenuItemDisabled(theme)
+                        : {}),
+                    }}
+                    >
+                    Reset to Default
+                  </DropdownItem>
+                </Dropdown>
+              </Row>
+
+              <EntityPageHorizontalTable
+                data={tableData.map(tableRow => Object.assign(
+                  {},
+                  tableRow,
+                  { key: tableRow.groupName || tableRow.key }
```

Comment:
Should use 
`key: tableRow.groupName !== undefined ? tableRow.groupName : tableRow.key,`
When tableRow.groupName equals to '', it means this tableRow is hidden. And if tableRow.groupName equals to undefined, it means this tableRow needs to be assigned a groupName with tableRow.key.